docs/adding_packages/sources_and_patches.md

```diff
@@ -32,9 +30,8 @@ Where ever possible, downloading source files and compiling is mandated. Downloa
 Downloaded source code must have a deterministic results where the exact same archive is download each time. See
 [Conandata's `"sha"` fields](conandata_yml_format.md#sha256) for how this is achieved in ConanCenterIndex.
 
-The sources stored under `self.source_folder` should not be modified. This will enable local workflows to "keep sources" and avoid extra downloads.
-Any patch should be applied to the copy of this source code when a build is executed (basically in `build()` method). See [Applying Patches](#applying-patches)
-below for more information.
+The sources stored under `self.source_folder` should not be modified, except for applying patches.
+Any patch should be applied to the copy of this source code when a source is executed (basically in `source()` method).
```

Comment:
```suggestion
Patches should be applied in the `source()` method - taking special care that the patches are platform agnostic. Patches in the `build()` method can be considered where this is not possible, provided that `no_copy_source` is **not** set to `True`.
```

although the complexity here is such that all of this should be covered in the Conan documentation and we should not have to restate it.

Reply:
Addressed by 237651b6a25001a267388afc785de60bdca2d1d7